transmogrifier/sources/transformer.py

```diff
@@ -72,6 +78,15 @@ def __next__(self) -> timdex.TimdexRecord:
             self.transformed_record_count += 1
             return record
 
+    def set_run_id(self, run_id: str | None) -> str:
+        """Method to set run_id for Transmogrifier run."""
+        if not run_id:
+            logger.info("explicit run_id not passed, minting new UUID")
+            run_id = str(uuid.uuid4())
```

Comment:
This, specifically, it what keeps this backwards compatible.  If a `run_id` is not passed via the CLI, which our current StepFunction will not do, a UUID is minted... and then just never used in any meaningful way.